### PR TITLE
Workaround for Cray compiler bug involving `NULL()` intrinsic

### PR DESCRIPTION
**Pull request by J-Lentz**
**Description**
The Cray compiler contains a bug where the `NULL()` intrinsic fails to produce a pointer of the correct type when it is passed directly to a subroutine expecting a pointer to a derived type.

This PR implements a workaround where `NULL()` is assigned to a variable which is passed to the subroutine, rather than calling `NULL()` directly as a subroutine argument.

**How Has This Been Tested?**
Builds with Cray compiler 15.0.1 on C5.

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included
- [x] `make distcheck` passes

### CODE REVIEW

**Review by laurenchilutti — APPROVED** [review]

**Comment by rem1776** [comment]
@J-Lentz Just a heads up, I'm going to push an empty commit to this PR to re-trigger the CI. It looks like it only failed due to #1480.

**Comment by J-Lentz** [comment]
This bug should be fixed in the next CCE release in a few months, so perhaps this PR doesn't need to be merged.

**Comment by laurenchilutti** [comment]
We should reopen this PR

**Review by rem1776 — DISMISSED** [review]

**Review by bensonr — COMMENTED** [review]
Will approve once the change requested by @uramirez8707 is addressed.

File: diag_manager/fms_diag_object.F90 [hunk 1]
```diff
@@ -224,6 +224,7 @@ integer function fms_register_diag_field_obj &
  integer, allocatable :: file_ids(:) !< The file IDs for this variable
  integer :: i !< For do loops
  integer, allocatable :: diag_field_indices(:) !< indices where the field was found in the yaml
+ class(diagDomain_t), pointer :: null_ptr => NULL() !< Workaround for a Cray compiler bug
```
[the remark this answers is not in the document]
Comment (J-Lentz) [reply diag_manager/fms_diag_object.F90]: Fixed in eb10e1876246ec4636bbf2967e769146352a39fd.

**Review by uramirez8707 — APPROVED** [review]

**Review by rem1776 — APPROVED** [review]

**Pull request merged**